scheduler/src/cook/mesos/optimizer.clj

```diff
@@ -19,122 +19,251 @@
             [clj-http.client :as http]
             [clj-time.core :as t]
             [clj-time.coerce :as tc]
-            [clj-time.periodic :as periodic]
-            [clojure.core.async :as async]
             [clojure.tools.logging :as log]
-            [cook.util :refer [lazy-load-var PosNum PosInt NonNegInt]]
+            [cook.util :refer [lazy-load-var NonNegInt PosNum PosInt]]
             [cook.mesos.util :as util]
-            [datomic.api :as d :refer (q)]
-            [schema.core :as s]))
+            [metrics.timers :as timers]
+            [plumbing.core :as pc]
+            [schema.core :as s])
+  (:import (java.util UUID)))
 
 (def TimePeriodMs NonNegInt)
 
+
+;; Elements missing:
+;;   1. Attributes (chip set, az, ..) -- have a combinatorial problem here
+;;   2. Price
+;;   3. Prob preemption in 10 minutes increments (or something like that)
```

Comment:
Should I understand this comment?
Are you going to remove it before merging?
I feel like we need some extra context here, or at very least a `TODO` or `FIXME` to clarify whether this is documentation or some kind of todo notes.